src/sinks/aws_cloudwatch_logs/config.rs

```diff
@@ -85,20 +90,46 @@ impl CloudwatchLogsSinkConfig {
         )
         .await
     }
+
+    pub async fn create_smithy_client(
+        &self,
+        proxy: &ProxyConfig,
+    ) -> crate::Result<aws_smithy_client::Client> {
+        let region = match self.region.region() {
+            Some(region) => Ok(region),
+            None => aws_config::default_provider::region::default_provider()
+                .region()
+                .await
+                .ok_or("Could not determine region from Vector configuration or default providers"),
+        }?;
+        create_smithy_client::<CloudwatchLogsClientBuilder>(
+            region,
+            proxy,
+            &self.tls,
+            true,
+            RetryConfig::disabled(),
+        )
+        .await
+    }
 }
 
 #[async_trait::async_trait]
 #[typetag::serde(name = "aws_cloudwatch_logs")]
 impl SinkConfig for CloudwatchLogsSinkConfig {
     async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
         let batcher_settings = self.batch.into_batcher_settings()?;
-        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
+        let request_settings = self
+            .request
+            .tower
+            .unwrap_with(&TowerRequestConfig::default());
         let client = self.create_client(cx.proxy()).await?;
+        let smithy_client = self.create_smithy_client(cx.proxy()).await?;
```

Comment:
It would be nice if the 2 `smithy_client`s being made here could be shared, but I'm not seeing a reasonable way to do that.